lib/apiservers/service/restapi/handlers/vch_create.go

```diff
@@ -0,0 +1,652 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/go-units"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/strfmt"
+	"gopkg.in/urfave/cli.v1"
+
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/cmd/vic-machine/common"
+	"github.com/vmware/vic/cmd/vic-machine/create"
+	"github.com/vmware/vic/lib/apiservers/service/models"
+	"github.com/vmware/vic/lib/apiservers/service/restapi/handlers/util"
+	"github.com/vmware/vic/lib/apiservers/service/restapi/operations"
+	"github.com/vmware/vic/lib/install/data"
+	"github.com/vmware/vic/lib/install/management"
+	"github.com/vmware/vic/lib/install/validate"
+	"github.com/vmware/vic/lib/install/vchlog"
+	"github.com/vmware/vic/pkg/ip"
+	viclog "github.com/vmware/vic/pkg/log"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/version"
+)
+
+const (
+	logFile = "vic-machine.log" // name of local log file
+)
+
+// VCHCreate is the handler for creating a VCH
+type VCHCreate struct {
+}
+
+// VCHCreate is the handler for creating a VCH within a Datacenter
+type VCHDatacenterCreate struct {
+}
+
+func (h *VCHCreate) Handle(params operations.PostTargetTargetVchParams, principal interface{}) middleware.Responder {
+	op := trace.NewOperation(params.HTTPRequest.Context(), "VCHCreate")
+
+	// Set up VCH create logger
+	localLogFile := setUpLogger()
+	// Close the two logging streams when done
+	defer vchlog.Close()
+	defer localLogFile.Close()
+
+	b := buildDataParams{
+		target:     params.Target,
+		thumbprint: params.Thumbprint,
+	}
+
+	d, err := buildData(op, b, principal)
+	if err != nil {
+		return operations.NewPostTargetTargetVchDefault(util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
```

Comment:
It's the name `go-swagger` generates; it basically just takes the path and concatenates the words in it (so `/target/{target}/vch` becomes `TargetTargetVch`). (Changing the name of the placeholders to avoid the "stutter" isn't the right answer.)

As far as I can tell, there's no way to suggest a better name or naming convention to `go-swagger`.